test_rmw_implementation/test/test_serialize_deserialize.cpp

```diff
@@ -112,6 +117,61 @@ TEST_F(CLASSNAME(TestSerializeDeserialize, RMW_IMPLEMENTATION), clean_round_trip
+    ROSIDL_GET_MSG_TYPE_SUPPORT(test_msgs, msg, BoundedSequencesNoStrings)};
+  test_msgs__msg__BoundedSequencesNoStrings input_message{};
+  test_msgs__msg__BoundedSequencesNoStrings output_message{};
+  ASSERT_TRUE(test_msgs__msg__BoundedSequencesNoStrings__init(&input_message));
+  ASSERT_TRUE(test_msgs__msg__BoundedSequencesNoStrings__init(&output_message));
+  rcutils_allocator_t default_allocator = rcutils_get_default_allocator();
+  rmw_serialized_message_t serialized_message = rmw_get_zero_initialized_serialized_message();
+  ASSERT_EQ(
+    RMW_RET_OK, rmw_serialized_message_init(
+      &serialized_message, 0lu, &default_allocator)) << rmw_get_error_string().str;
+
+  // Make input_message not equal to output_message.
+  ASSERT_TRUE(rosidl_runtime_c__bool__Sequence__init(&input_message.bool_values, 1));
+  input_message.bool_values.data[0] = true;
+  ASSERT_TRUE(rosidl_runtime_c__int16__Sequence__init(&input_message.int16_values, 1));
+  input_message.int16_values.data[0] = -7;
+
+  rmw_ret_t ret = rmw_serialize(&input_message, ts, &serialized_message);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  EXPECT_NE(nullptr, serialized_message.buffer);
+  EXPECT_GT(serialized_message.buffer_length, 0lu);
+
+  // Adding more items should increase buffer length
+  auto first_message_length = serialized_message.buffer_length;
+  ASSERT_TRUE(rosidl_runtime_c__int32__Sequence__init(&input_message.int32_values, 1));
+  input_message.int32_values.data[0] = -1;
+  ASSERT_TRUE(rosidl_runtime_c__uint16__Sequence__init(&input_message.uint16_values, 1));
+  input_message.uint16_values.data[0] = 125;
+  
+  ret = rmw_serialize(&input_message, ts, &serialized_message);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  EXPECT_NE(nullptr, serialized_message.buffer);
+  EXPECT_GT(serialized_message.buffer_length, 0lu);
+  EXPECT_GT(serialized_message.buffer_length, first_message_length);
+
+  ret = rmw_deserialize(&serialized_message, ts, &output_message);
+  EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
+  EXPECT_EQ(input_message.bool_values.size, output_message.bool_values.size);
+  EXPECT_EQ(input_message.bool_values.data[0], output_message.bool_values.data[0]);
+  EXPECT_EQ(input_message.int16_values.size, output_message.int16_values.size);
+  EXPECT_EQ(input_message.int16_values.data[0], output_message.int16_values.data[0]);
+  EXPECT_EQ(input_message.int32_values.size, output_message.int32_values.size);
+  EXPECT_EQ(input_message.int32_values.data[0], output_message.int32_values.data[0]);
+  EXPECT_EQ(input_message.uint16_values.size, output_message.uint16_values.size);
+  EXPECT_EQ(input_message.uint16_values.data[0], output_message.uint16_values.data[0]);
+
+  test_msgs__msg__BoundedSequencesNoStrings__fini(&input_message);
+  test_msgs__msg__BoundedSequencesNoStrings__fini(&output_message);
+
+  EXPECT_EQ(RMW_RET_OK, rmw_serialized_message_fini(&serialized_message)) <<
+    rmw_get_error_string().str;
```

Comment:
@MiguelCompany consider moving these lines to `OSRF_TESTING_SCOPE_EXIT` blocks.

Reply:
> Why wouldn't test_msgs/msg/BoundedSequences work?

Because it has strings, which are unbounded

> it doesn't seem like `test_msgs/msg/UnboundedSequencesNoStrings` is used at all

I just added it for symmetry